common/src/View/UVViewHelper.cpp

```diff
@@ -128,20 +128,29 @@ namespace TrenchBroom {
 
             const auto* texture = m_face->texture();
             if (texture != nullptr) {
-
                 const auto& boundary = m_face->boundary();
-                const auto distance = vm::intersect_ray_plane(ray, boundary);
-                const auto hitPointInWorldCoords = vm::point_at_distance(ray, distance);
-                const auto hitPointInTexCoords = m_face->toTexCoordSystemMatrix(m_face->offset(), m_face->scale(), true) * hitPointInWorldCoords;
+                const FloatType distance = vm::intersect_ray_plane(ray, boundary);
+                const vm::vec3  hitPointInWorldCoords = vm::point_at_distance(ray, distance);
+                const vm::vec3  hitPointInTexCoords = m_face->toTexCoordSystemMatrix(m_face->offset(), m_face->scale(), true) * hitPointInWorldCoords;
+                const vm::vec2f hitPointInViewCoords = m_camera.project(vm::vec3f(hitPointInWorldCoords)).xy();
+
+                const vm::vec2f closestGridInTexCoords = vm::vec2f(hitPointInTexCoords) + computeDistanceFromTextureGrid(hitPointInTexCoords);
+                const vm::vec3  closestGridInWorldCoords = m_face->fromTexCoordSystemMatrix(m_face->offset(), m_face->scale(), true) * vm::vec3(closestGridInTexCoords, 0.0f);
+                const vm::vec2f closestGridInViewCoords = m_camera.project(vm::vec3f(closestGridInWorldCoords)).xy();
 
-                const auto maxDistance = static_cast<FloatType>(5.0) / static_cast<FloatType>(cameraZoom());
-                const auto stripeSize = UVViewHelper::stripeSize();
+                // X and Y distance in pixels to the closest grid intersection
+                // (i.e. so the X component is the distance from the closest vertical gridline, and the
+                // Y the distance to the closest horizontal gridline.)
+                // FIXME: should be measured in points so the grid isn't harder to hit with high-DPI
+                const vm::vec2f distToClosestGridInScreenCoords = vm::abs(closestGridInViewCoords - hitPointInViewCoords);
 
                 for (size_t i = 0; i < 2; ++i) {
-                    const auto closestStrip = vm::snap(hitPointInTexCoords[i], stripeSize[i]);
-                    const auto error = vm::abs(hitPointInTexCoords[i] - closestStrip);
+                    const float maxDistance = static_cast<float>(5.0);
```

Comment:
Make this constexpr and move it out of the loop? Maybe that should even be a constant or preference, so that we can make it coherent across tools. I think there is similar logic in the vertex tool where it allows you to click outside of the vertex handle by a certain distance and it will still register.

Reply:
sounds good, for now I left it as a fixme

Reply:
ok!